### PR TITLE
fix(dockerfile): handle quoted absolute path in CKV_DOCKER_10 

### LINKED ISSUE #3625: CKV_DOCKER_10 mistakes quoted absolute paths for relative paths
**Describe the issue**
CKV_DOCKER_10 mistakes quoted absolute paths for relative paths.

**Examples**
```
cat << EOF > Dockerfile
FROM alpine:3.16
WORKDIR "/app"
EOF

checkov --check CKV_DOCKER_10 --file Dockerfile
```

image

**Version (please complete the following information):**
2.1.258

### PR DESCRIPTION
**Pull request by ttd2089**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

Updates CKV_DOCKER_10 to recognize optional quotes around paths following the `WORKDIR` instruction.

Fixes #3625

## Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by ttd2089** [comment]
> **By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**
> ## Description
> 
> Updates CKV_DOCKER_10 to recognize optional quotes around paths following the `WORKDIR` instruction.
> 
> Fixes #3625
> ## Checklist:
> 
>     * [ ]  My code follows the style guidelines of this project
> 
>     * [x]  I have performed a self-review of my own code
> 
>     * [x]  I have commented my code, particularly in hard-to-understand areas
> 
>     * [ ]  I have made corresponding changes to the documentation
> 
>     * [x]  I have added tests that prove my feature, policy, or fix is effective and works
> 
>     * [x]  New and existing tests pass locally with my changes
> 
>     * [x]  Any dependent changes have been merged and published in downstream modules

I left the style and documentation items unchecked because I would defer to the maintainers to determine if their requirements are met.

**Review by gruebel — APPROVED** [review]
nice, thanks for the contribution 🚀 could you onetime merge master into your branch?

**Review by tronxd — APPROVED** [review]

**Pull request merged**
